### DISCUSSION TITLE
v3.5.0

### DISCUSSION THREAD

**Discussion by nafg** [slick/slick#2908] [Announcements]
### Highlights 🎉
* Drop support for old SQL Server versions by @nafg in https://github.com/slick/slick/pull/2558
* Insert multiple rows using a single SQL statement by @jkugiya in https://github.com/slick/slick/pull/2398
* Adds insertOrUpdateAll by @jkugiya in https://github.com/slick/slick/pull/2397
* Fix SQL Server tests. Drops JTDS by @nafg in https://github.com/slick/slick/pull/2560
* Drop scala 2.12 by @nafg in https://github.com/slick/slick/pull/2633
* Add -Xsource:3 to scalacOptions by @nafg in https://github.com/slick/slick/pull/2662
* Remove some more advanced HList features by @nafg in https://github.com/slick/slick/pull/2693
* Remove specialization by @nafg in https://github.com/slick/slick/pull/2695
* Remove `MappedTo` trait (single-param case classes extending it were mapped automatically to the underlying type) by @nafg in https://github.com/slick/slick/pull/2698
* Remove `tsql` interpolator, and `stripMargin` method for Plain SQL queries. by @nafg in https://github.com/slick/slick/pull/2696
* un-deprecate `countDistinct` method by @xuwei-k in https://github.com/slick/slick/pull/2649
* Remove deprecated features by @nafg in https://github.com/slick/slick/pull/2710
* Revert drop scala 212 by @mdedetrich in https://github.com/slick/slick/pull/2734
* #2701 Add conditional filterNot methods (filterNotIf/filterNotOpt) support by @sebastianrozyczka in https://github.com/slick/slick/pull/2751
* Scala 3 support by @nafg in https://github.com/slick/slick/pull/2687
### Enhancements ✨
* Attach base statement as child for asTry dumpInfo by @natefitzgerald in https://github.com/slick/slick/pull/2584
* More type-safe way of expressing profiles' support of RowsPerStatement.All by @nafg in https://github.com/slick/slick/pull/2612
* Add exceptionOverrideClassName config for HikariCP by @jsoizo in https://github.com/slick/slick/pull/2621
* Make `b` interpolator not use macros by @nafg in https://github.com/slick/slick/pull/2684
* Restore ability to concat sql strings, closes #1161 by @OlegYch in https://github.com/slick/slick/pull/2796
* MappedProjection: make the source type no longer part of the API by @nafg in https://github.com/slick/slick/pull/2854
* Fix `getOrElse` type inference issue in Scala 3, closes #2761 by @OlegYch in https://github.com/slick/slick/pull/2803
### Bugs fixed 🐛️
* Handle in-use count errors by @nafg in https://github.com/slick/slick/pull/2619
* BasicBackend#DatabaseDef#releaseSession: null out the session first, … by @nafg in https://github.com/slick/slick/pull/2620
* Postgres date and time boundary conditions by @JamesMcIntosh in https://github.com/slick/slick/pull/2643
* avoid deprecated `java.net.URL` constructor by @xuwei-k in https://github.com/slick/slick/pull/2691
* OracleProfile: restore parentheses by @nafg in https://github.com/slick/slick/pull/2697
* For some reason searching by dates in some timezones occasionally fails on Oracle 11g XE, avoid them in tests by @OlegYch in https://github.com/slick/slick/pull/2685
* Fixed behavior of `insertOrUpdate` on tables with auto_increment option by @jkugiya in https://github.com/slick/slick/pull/2785
* Replace filter on aggregates with CASE, closes #2235 by @OlegYch in https://github.com/slick/slick/pull/2832
* Support referencing group keys in nested query, closes #2422 by @OlegYch in https://github.com/slick/slick/pull/2877
### Documentation improvements 📃️
* Document using the generic jdbc profile by @ElectricWound in https://github.com/slick/slick/pull/2586
* build.sbt: set site description setting by @nafg in https://github.com/slick/slick/pull/2623
* Document tupled workaround for Scala 3 by @mdedetrich in https://github.com/slick/slick/pull/2760
* Fix scaladoc -sourcepath issues by @nafg in https://github.com/slick/slick/pull/2769
* schemas.md: improve Mapped Tables section by @nafg in https://github.com/slick/slick/pull/2770
* upgrade.md: Add initial 3.5.0 section by @nafg in https://github.com/slick/slick/pull/2773
* upgrade.md: fix sections by @nafg in https://github.com/slick/slick/pull/2805
* upgrade.md: mention the change to MappedProjection from PR #2854 by @nafg in https://github.com/slick/slick/pull/2860
### Chores 🛠
* Update slf4j-api to 2.0.0 by @scala-steward in https://github.com/slick/slick/pull/2549
* Update slf4j-api to 2.0.1 by @scala-steward in https://github.com/slick/slick/pull/2554
* Update dependency com.typesafe:sbt-mima-plugin to v1.1.1 by @renovate in https://github.com/slick/slick/pull/2555
* Drop Java 8 support by @nafg in https://github.com/slick/slick/pull/2557
* CompatReportPlugin: use details/summary tags by @nafg in https://github.com/slick/slick/pull/2559
* Update sbt-scoverage to 2.0.3 by @scala-steward in https://github.com/slick/slick/pull/2553
* Update logback-classic to 1.4.1 by @scala-steward in https://github.com/slick/slick/pull/2552
* Update dependency com.zaxxer:HikariCP to v5 by @renovate in https://github.com/slick/slick/pull/2463
* Update scala-compiler, scala-library to 2.12.17 by @scala-steward in https://github.com/slick/slick/pull/2556
* Update dependency org.scoverage:sbt-scoverage to v2.0.4 by @renovate in https://github.com/slick/slick/pull/2561
* Update dependency org.slf4j:slf4j-api to v2.0.2 by @renovate in https://github.com/slick/slick/pull/2562
* Update dependency org.slf4j:slf4j-api to v2.0.3 by @renovate in https://github.com/slick/slick/pull/2563
* build.sbt: versionPolicyIntention: BumpMajor by @nafg in https://github.com/slick/slick/pull/2565
* build.sbt: testAll should run versionPolicyCheck on actual modules by @nafg in https://github.com/slick/slick/pull/2566
* Update dependency org.scalatestplus:testng-7-5 to v3.2.14.0 by @renovate in https://github.com/slick/slick/pull/2564
* Update dependency ch.qos.logback:logback-classic to v1.4.2 by @renovate in https://github.com/slick/slick/pull/2568
* Update dependency ch.qos.logback:logback-classic to v1.4.3 by @renovate in https://github.com/slick/slick/pull/2569
* Update dependency ch.qos.logback:logback-classic to v1.4.4 by @renovate in https://github.com/slick/slick/pull/2574
* Update dependency org.scoverage:sbt-scoverage to v2.0.5 by @renovate in https://github.com/slick/slick/pull/2576
* Update hsqldb to 2.7.0 by @scala-steward in https://github.com/slick/slick/pull/2511
* Update sbt to 1.7.2 by @scala-steward in https://github.com/slick/slick/pull/2582
* Update ibmcom/db2 Docker tag to v11.5.8.0 by @renovate in https://github.com/slick/slick/pull/2585
* Update mysql-connector-java to 8.0.31 by @scala-steward in https://github.com/slick/slick/pull/2580
* Update dependency org.scoverage:sbt-scoverage to v2.0.6 by @renovate in https://github.com/slick/slick/pull/2587
* .mergify.yml: don't require signing Lightbend CLA by @nafg in https://github.com/slick/slick/pull/2590
* Update dependency org.hsqldb:hsqldb to v2.7.1 by @renovate in https://github.com/slick/slick/pull/2592
* Use scala version wildcards for ci.yml matrix by @nafg in https://github.com/slick/slick/pull/2594
* Update sbt-ci-release to 1.5.11 by @scala-steward in https://github.com/slick/slick/pull/2593
* Update scala-compiler, scala-library to 2.13.10 by @slick-scala-steward in https://github.com/slick/slick/pull/2577
* Update paradox-theme-generic, sbt-paradox to 0.10.3 by @slick-scala-steward in https://github.com/slick/slick/pull/2596
* Update dependency ch.epfl.scala:sbt-version-policy to v2.1.0 by @renovate in https://github.com/slick/slick/pull/2597
* use `CollectionConverters` instead of deprecated `JavaConverters` by @xuwei-k in https://github.com/slick/slick/pull/2599
* add explicit types for implicits by @xuwei-k in https://github.com/slick/slick/pull/2598
* Update sbt to 1.7.3 by @slick-scala-steward in https://github.com/slick/slick/pull/2600
* Revert "Update sbt to 1.7.3" by @nafg in https://github.com/slick/slick/pull/2601
* Update sqlite-jdbc to 3.39.4.0 by @slick-scala-steward in https://github.com/slick/slick/pull/2603
* Update sbt to 1.7.3 by @scala-steward in https://github.com/slick/slick/pull/2604
* Update sbt to 1.8.0 by @slick-scala-steward in https://github.com/slick/slick/pull/2605
* Update dependency org.xerial:sqlite-jdbc to v3.39.4.1 by @renovate in https://github.com/slick/slick/pull/2606
* Update dependency org.slf4j:slf4j-api to v2.0.4 by @renovate in https://github.com/slick/slick/pull/2607
* Update dependency ch.qos.logback:logback-classic to v1.4.5 by @renovate in https://github.com/slick/slick/pull/2608
* Update dependency org.xerial:sqlite-jdbc to v3.40.0.0 by @renovate in https://github.com/slick/slick/pull/2609
* Update dependency org.postgresql:postgresql to v42.5.1 by @renovate in https://github.com/slick/slick/pull/2610
* Update dependency org.slf4j:slf4j-api to v2.0.5 by @renovate in https://github.com/slick/slick/pull/2611
* Update scala-collection-compat to 2.9.0 by @slick-scala-steward in https://github.com/slick/slick/pull/2613
* pr-compat-report-generate.yml: use -no-colors and switch to GH Environment Files by @nafg in https://github.com/slick/slick/pull/2616
* pr-compat-report-generate.yml: give up on piping and do the writing within SBT by @nafg in https://github.com/slick/slick/pull/2617
* Update dependency com.oracle.database.jdbc.debug:ojdbc8_g to v21.8.0.0 by @renovate in https://github.com/slick/slick/pull/2618
* Update dependency org.slf4j:slf4j-api to v2.0.6 by @renovate in https://github.com/slick/slick/pull/2624
* Restore Java 8 support by @nafg in https://github.com/slick/slick/pull/2622
* Update dependency org.scalatestplus:testng-7-5 to v3.2.15.0 by @renovate in https://github.com/slick/slick/pull/2629
* Update sbt to 1.8.2 by @scala-steward in https://github.com/slick/slick/pull/2630
* Publish with JDK 8 again by @nafg in https://github.com/slick/slick/pull/2632
* Update dependency org.freemarker:freemarker to v2.3.32 by @renovate in https://github.com/slick/slick/pull/2636
* Update dependency mysql:mysql-connector-java to v8.0.32 by @renovate in https://github.com/slick/slick/pull/2637
* Update dependency org.postgresql:postgresql to v42.5.2 by @renovate in https://github.com/slick/slick/pull/2644
* Update dependency org.xerial:sqlite-jdbc to v3.40.1.0 by @renovate in https://github.com/slick/slick/pull/2646
* Update dependency org.postgresql:postgresql to v42.5.3 by @renovate in https://github.com/slick/slick/pull/2647
* Update dependency org.scoverage:sbt-scoverage to v2.0.7 by @renovate in https://github.com/slick/slick/pull/2650
* Update dependency com.oracle.database.jdbc.debug:ojdbc8_g to v21.9.0.0 by @renovate in https://github.com/slick/slick/pull/2651
* Update dependency org.postgresql:postgresql to v42.5.4 by @renovate in https://github.com/slick/slick/pull/2652
* Fix compiler deprecated warnings by @slovvik in https://github.com/slick/slick/pull/2529
* Update dependency org.xerial:sqlite-jdbc to v3.41.0.0 by @renovate in https://github.com/slick/slick/pull/2653
* Remove scala-collection-compat by @nafg in https://github.com/slick/slick/pull/2656
* Update sbt-sonatype by @nafg in https://github.com/slick/slick/pull/2666
* Update dependency org.xerial.sbt:sbt-sonatype to v3.9.18 by @renovate in https://github.com/slick/slick/pull/2675
* Update logback-classic to 1.3.6 by @scala-steward in https://github.com/slick/slick/pull/2677
* Update dependency org.slf4j:slf4j-api to v2.0.7 by @renovate in https://github.com/slick/slick/pull/2679
* Update dependency org.postgresql:postgresql to v42.6.0 by @renovate in https://github.com/slick/slick/pull/2680
* Avoid compiling entire project on Intellij import by @OlegYch in https://github.com/slick/slick/pull/2678
* Update dependency com.typesafe:sbt-mima-plugin to v1.1.2 by @renovate in https://github.com/slick/slick/pull/2699
* Update sbt-sdlc to 0.3.0 by @slick-scala-steward in https://github.com/slick/slick/pull/2704
* Update dependency mysql:mysql-connector-java to v8.0.33 by @renovate in https://github.com/slick/slick/pull/2706
* Update logback-classic to 1.3.7 by @slick-scala-steward in https://github.com/slick/slick/pull/2707
* Update dependency org.xerial.sbt:sbt-sonatype to v3.9.19 by @renovate in https://github.com/slick/slick/pull/2712
* Update sbt-sonatype to 3.9.19 by @slick-scala-steward in https://github.com/slick/slick/pull/2713
* Update sbt-ci-release to 1.5.12 by @slick-scala-steward in https://github.com/slick/slick/pull/2715
* Update dependency org.xerial.sbt:sbt-sonatype to v3.9.20 by @renovate in https://github.com/slick/slick/pull/2716
* Update sbt to 1.8.3 by @slick-scala-steward in https://github.com/slick/slick/pull/2718
* Update dependency org.scalatestplus:testng-7-5 to v3.2.16.0 by @renovate in https://github.com/slick/slick/pull/2721
* Update sbt-sonatype to 3.9.21 by @slick-scala-steward in https://github.com/slick/slick/pull/2722
* Update ojdbc8_g to 21.10.0.0 by @slick-scala-steward in https://github.com/slick/slick/pull/2723
* Update sbt to 1.9.0 by @slick-scala-steward in https://github.com/slick/slick/pull/2728
* Update dependency org.scoverage:sbt-scoverage to v2.0.8 by @renovate in https://github.com/slick/slick/pull/2730
* Update sbt-scoverage to 2.0.8 by @slick-scala-steward in https://github.com/slick/slick/pull/2729
* Update dependency ch.epfl.scala:sbt-version-policy to v2.1.1 by @renovate in https://github.com/slick/slick/pull/2732
* Update scala-compiler, scala-library to 2.13.11 by @slick-scala-steward in https://github.com/slick/slick/pull/2737
* Update logback-classic to 1.3.8 by @slick-scala-steward in https://github.com/slick/slick/pull/2740
* Update scala-compiler, scala-library to 2.12.18 by @slick-scala-steward in https://github.com/slick/slick/pull/2741
* Update actions/upload-artifact action to v3 by @renovate in https://github.com/slick/slick/pull/2744
* Update dependency org.scala-lang.modules:scala-collection-compat to v2.11.0 by @renovate in https://github.com/slick/slick/pull/2749
* Update scala-collection-compat to 2.11.0 by @slick-scala-steward in https://github.com/slick/slick/pull/2748
* Update dependency sbt/sbt to v1.9.1 by @renovate in https://github.com/slick/slick/pull/2752
* Update mysql client dependency by @xuwei-k in https://github.com/slick/slick/pull/2754
* Update sbt to 1.9.2 by @slick-scala-steward in https://github.com/slick/slick/pull/2757
* fix warning in `TransactionTest` by @xuwei-k in https://github.com/slick/slick/pull/2762
* Update dependency com.lightbend.paradox:sbt-paradox to v0.10.4 by @renovate in https://github.com/slick/slick/pull/2764
* Update dependency com.lightbend.paradox:sbt-paradox to v0.10.5 by @renovate in https://github.com/slick/slick/pull/2765
* Update paradox-theme-generic, sbt-paradox to 0.10.5 by @slick-scala-steward in https://github.com/slick/slick/pull/2766
* Try Scala 3.3.1-RC3 by @nafg in https://github.com/slick/slick/pull/2758
* Update scala3-library to 3.3.1-RC4 by @slick-scala-steward in https://github.com/slick/slick/pull/2768
* build.sbt: default scala version to 2.13 by @nafg in https://github.com/slick/slick/pull/2771
* CompatReportPlugin: Remove `<detail>/<summary>` by @nafg in https://github.com/slick/slick/pull/2772
* build.sbt: buildCompatReport: include header by @nafg in https://github.com/slick/slick/pull/2775
* Update dependency com.mysql:mysql-connector-j to v8.1.0 by @renovate in https://github.com/slick/slick/pull/2777
* Update sbt-version-policy to 2.1.2 by @slick-scala-steward in https://github.com/slick/slick/pull/2778
* Update dependency com.typesafe:sbt-mima-plugin to v1.1.3 by @renovate in https://github.com/slick/slick/pull/2780
* Update sbt-mima-plugin to 1.1.3 by @slick-scala-steward in https://github.com/slick/slick/pull/2781
* Update dependency sbt/sbt to v1.9.3 by @renovate in https://github.com/slick/slick/pull/2786
* Update sbt-version-policy to 2.1.3 by @slick-scala-steward in https://github.com/slick/slick/pull/2787
* Update logback-classic to 1.3.9 by @slick-scala-steward in https://github.com/slick/slick/pull/2789
* Update logback-classic to 1.3.10 by @slick-scala-steward in https://github.com/slick/slick/pull/2792
* Update logback-classic to 1.3.11 by @slick-scala-steward in https://github.com/slick/slick/pull/2794
* Update scala3-library to 3.3.1-RC5 by @slick-scala-steward in https://github.com/slick/slick/pull/2797
* Update ojdbc8_g to 21.11.0.0 by @slick-scala-steward in https://github.com/slick/slick/pull/2798
* Explicitly handle this.type soundness reported as error since Scala 3.4.x by @WojciechMazur in https://github.com/slick/slick/pull/2801
* Only use scala-collection-compat for Scala 2.12 by @mdedetrich in https://github.com/slick/slick/pull/2747
* Fix incorrect POM filter with case sensitivity by @mdedetrich in https://github.com/slick/slick/pull/2804
* Update scala3-library to 3.3.1-RC6 by @slick-scala-steward in https://github.com/slick/slick/pull/2806
* Add scalac inliner options for Scala 2 by @mdedetrich in https://github.com/slick/slick/pull/2808
* Update sbt to 1.9.4 by @slick-scala-steward in https://github.com/slick/slick/pull/2812
* Update scala3-library to 3.3.1-RC7 by @slick-scala-steward in https://github.com/slick/slick/pull/2813
* Update dependency org.slf4j:slf4j-api to v2.0.8 by @renovate in https://github.com/slick/slick/pull/2817
* Update slf4j-api to 2.0.8 by @slick-scala-steward in https://github.com/slick/slick/pull/2816
* Update dependency org.slf4j:slf4j-api to v2.0.9 by @renovate in https://github.com/slick/slick/pull/2819
* Update actions/checkout action to v4 by @renovate in https://github.com/slick/slick/pull/2820
* Update dependency org.scalatestplus:testng-7-5 to v3.2.17.0 by @renovate in https://github.com/slick/slick/pull/2821
* Update scala3-library to 3.3.1 by @slick-scala-steward in https://github.com/slick/slick/pull/2823
* Update dependency org.scoverage:sbt-scoverage to v2.0.9 by @renovate in https://github.com/slick/slick/pull/2824
* add explicit type by @xuwei-k in https://github.com/slick/slick/pull/2825
* Update dependency sbt/sbt to v1.9.5 by @renovate in https://github.com/slick/slick/pull/2829
* Update sbt to 1.9.5 by @slick-scala-steward in https://github.com/slick/slick/pull/2830
* Update dependency sbt/sbt to v1.9.6 by @renovate in https://github.com/slick/slick/pull/2833
* Update sbt to 1.9.6 by @slick-scala-steward in https://github.com/slick/slick/pull/2834
* Hide old compatibility changes report by @xuwei-k in https://github.com/slick/slick/pull/2840
* add explicit type by @xuwei-k in https://github.com/slick/slick/pull/2838
* Revert "Hide old compatibility changes report" by @nafg in https://github.com/slick/slick/pull/2841
* Test on JDK 21 instead of 17 and fix  by @xuwei-k in https://github.com/slick/slick/pull/2839
* Scala 2.13.12 by @xuwei-k in https://github.com/slick/slick/pull/2842
* Update typesafe:config to 1.4.3 by @slick-scala-steward in https://github.com/slick/slick/pull/2844
* Update dependency sbt/sbt to v1.9.7 by @renovate in https://github.com/slick/slick/pull/2846
* Update sbt to 1.9.7 by @slick-scala-steward in https://github.com/slick/slick/pull/2847
* Update dependency com.mysql:mysql-connector-j to v8.2.0 by @renovate in https://github.com/slick/slick/pull/2850
* Update actions/github-script action to v7 by @renovate in https://github.com/slick/slick/pull/2857
* Update dependency org.xerial.sbt:sbt-sonatype to v3.10.0 by @renovate in https://github.com/slick/slick/pull/2859
* Update dependency org.postgresql:postgresql to v42.7.0 by @renovate in https://github.com/slick/slick/pull/2861
* Update logback-classic to 1.3.12 by @scala-steward in https://github.com/slick/slick/pull/2864
* Update actions/setup-java action to v4 by @renovate in https://github.com/slick/slick/pull/2866
* Update logback-classic to 1.3.13 by @scala-steward in https://github.com/slick/slick/pull/2867
* Update logback-classic to 1.3.14 by @scala-steward in https://github.com/slick/slick/pull/2870
* Update dependency org.postgresql:postgresql to v42.7.1 by @renovate in https://github.com/slick/slick/pull/2873
* Update dependency sbt/sbt to v1.9.8 by @renovate in https://github.com/slick/slick/pull/2875
* Update dependency org.slf4j:slf4j-api to v2.0.10 by @renovate in https://github.com/slick/slick/pull/2881
* Update slf4j-api to 2.0.11 by @scala-steward in https://github.com/slick/slick/pull/2884
* Update dependency com.mysql:mysql-connector-j to v8.3.0 by @renovate in https://github.com/slick/slick/pull/2885
* Update dependency com.lightbend.paradox:sbt-paradox to v0.10.6 by @renovate in https://github.com/slick/slick/pull/2886
* Update ts-graphviz/setup-graphviz action to v2 by @renovate in https://github.com/slick/slick/pull/2887
* Update dependency org.slf4j:slf4j-api to v2.0.12 by @renovate in https://github.com/slick/slick/pull/2888
* Only run inliner in CI by @mdedetrich in https://github.com/slick/slick/pull/2889
* Update dependency org.scoverage:sbt-scoverage to v2.0.10 by @renovate in https://github.com/slick/slick/pull/2892
* Update dependency org.postgresql:postgresql to v42.7.2 by @renovate in https://github.com/slick/slick/pull/2895
* Update dependency sbt/sbt to v1.9.9 by @renovate in https://github.com/slick/slick/pull/2897
* Update ojdbc8_g to 21.13.0.0 by @scala-steward in https://github.com/slick/slick/pull/2898
* Update dependency org.scoverage:sbt-scoverage to v2.0.11 by @renovate in https://github.com/slick/slick/pull/2899
* Update scala-compiler, scala-library to 2.13.13 by @scala-steward in https://github.com/slick/slick/pull/2901
* Update scala-compiler, scala-library to 2.12.19 by @scala-steward in https://github.com/slick/slick/pull/2900
### Other Changes
* Remove scala.language.higherKinds import by @nafg in https://github.com/slick/slick/pull/2657
* Delete deprecated TestkitTest by @nafg in https://github.com/slick/slick/pull/2659
* Eliminate last warning by @nafg in https://github.com/slick/slick/pull/2660
* Fix scaladoc warnings by @nafg in https://github.com/slick/slick/pull/2661
* Fix mis-renamed SQLServerProfile#XXXJdbcType by @nafg in https://github.com/slick/slick/pull/2664
* Remove unused imports by @nafg in https://github.com/slick/slick/pull/2663
* Change sonatypeProfileName from com.typesafe to com.typesafe.slick by @nafg in https://github.com/slick/slick/pull/2667
* Some cleanups by @nafg in https://github.com/slick/slick/pull/2683
* An indentation fix by @nafg in https://github.com/slick/slick/pull/2692
* Line breaks after infix operator by @nafg in https://github.com/slick/slick/pull/2694
* OracleProfile: Delete unused and incomplete OracleLocalTimeJdbcType from #1872 by @nafg in https://github.com/slick/slick/pull/2686
* A few cleanups by @nafg in https://github.com/slick/slick/pull/2717
* Remove headOption extension method for ReadAheadIterator by @nafg in https://github.com/slick/slick/pull/2719
* Remove usage of reflection to support pre-JDBC 4.1 (pre-Java 7) by @nafg in https://github.com/slick/slick/pull/2720
* Some cleanup by @nafg in https://github.com/slick/slick/pull/2739

## New Contributors
* @ElectricWound made their first contribution in https://github.com/slick/slick/pull/2586
* @natefitzgerald made their first contribution in https://github.com/slick/slick/pull/2584
* @slick-scala-steward made their first contribution in https://github.com/slick/slick/pull/2577
* @jsoizo made their first contribution in https://github.com/slick/slick/pull/2621
* @slovvik made their first contribution in https://github.com/slick/slick/pull/2529
* @JamesMcIntosh made their first contribution in https://github.com/slick/slick/pull/2643
* @OlegYch made their first contribution in https://github.com/slick/slick/pull/2678
* @mdedetrich made their first contribution in https://github.com/slick/slick/pull/2734
* @sebastianrozyczka made their first contribution in https://github.com/slick/slick/pull/2751
* @WojciechMazur made their first contribution in https://github.com/slick/slick/pull/2801

**Full Changelog**: https://github.com/slick/slick/compare/v3.4.1...v3.5.0

<hr /><em>This discussion was created from the release <a href='https://github.com/slick/slick/releases/tag/v3.5.0'>v3.5.0</a>.</em>

**Comment by mkurz** [comment]
Thanks! New play-slick releases are out:
- https://github.com/playframework/play-slick/releases/tag/6.1.0
- https://github.com/playframework/play-slick/releases/tag/5.3.0

**Comment by mdedetrich** [comment]
Thanks for everything, made a discussion to release pekko-persistence-jdbc as a M1 just to check that everything is okay, see https://lists.apache.org/thread/95sjkpbdomwbbhvw1os9g0ol2f6rt32b

**Comment by amumurst** [comment]
https://github.com/tminglei/slick-pg/issues/703 Unsure why there is no PR made by scala-steward yet to update to 3.5.0, but perhaps that only runs once per day.
